programs/drift/src/math/margin.rs

```diff
@@ -484,6 +503,14 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
 
         calculation.add_total_collateral(weighted_pnl)?;
 
+        #[cfg(drift_rs)]
+        match weighted_pnl.cmp(&0) {
+            Ordering::Less => {
+                calculation.add_perp_liability_value(weighted_pnl.unsigned_abs())?;
```

Comment:
we need to add a separate field for perp pnl and perp libability

should review getLeverage in ts to see how that is used

Reply:
need to add worst_case_base_asset_value to perp liability